spec/vcr_setup.rb

```diff
@@ -0,0 +1,41 @@
+require 'vcr'
+require 'rspec'
+
+module BlockScore
+  VCR.configure do |c|
+    c.cassette_library_dir = 'spec/cassettes'
+    c.hook_into :webmock
+    c.configure_rspec_metadata!
+    c.ignore_hosts 'codeclimate.com'
+
+    c.filter_sensitive_data('BLOCKSCORE_TEST_KEY') { ENV['BLOCKSCORE_TEST_KEY'] }
+  end
+
+  RSpec.configure do |config|
+    config.before(:suite) do
+      BlockScore.api_key = ENV.fetch('BLOCKSCORE_TEST_KEY', 'BLOCKSCORE_TEST_KEY')
+    end
+
+    config.around(:each) do |example|
+      options = example.metadata[:vcr] || {
+        record: (ENV['VCR_RECORD_MODE'] ? ENV['VCR_RECORD_MODE'].to_sym : :none)
+      }
+
+      if options[:record] == :skip
+        VCR.turned_off(&example)
+      else
+        name = example.metadata[:full_description]
+          .split(/\s+/, 2)
+          .join('/')
+          .underscore
+          .gsub(/[.#?]/, '/')
+          .gsub(%r(\/+), '/')
+          .strip
+          .tr(' ', '_')
+          .gsub(%r([/_ ]$), '')
+
+        VCR.use_cassette(name, options, &example)
+      end
+    end
+  end
+end
```

Comment:
I didn't go as far as I usually would with modifying this, but this is sort of what I had in mind:

```ruby
require 'vcr'
require 'rspec'

module BlockScore
  RECORD_MODE_KEY     = 'VCR_RECORD_MODE'
  ENVIRONMENT_API_KEY = 'BLOCKSCORE_TEST_KEY'
  DEFAULT_API_KEY     = 'BLOCKSCORE_TEST_KEY'

  extend self

  RSpec.configure do |config|
    config.before(:suite) do
      BlockScore.api_key = ENV.fetch(ENVIRONMENT_API_KEY, DEFAULT_API_KEY)
    end

    config.around(:each) do |example|
      record_option = ENV.fetch(RECORD_MODE_KEY, :none).to_sym
      options = example.metadata.fetch(:vcr, record: record_option)
      run_vcr_example(options, example)
    end
  end

  VCR.configure do |c|
    c.cassette_library_dir = 'spec/cassettes'
    c.hook_into :webmock
    c.configure_rspec_metadata!
    c.ignore_hosts 'codeclimate.com'

    c.filter_sensitive_data(DEFAULT_API_KEY) { ENV[ENVIRONMENT_API_KEY] }
  end

  def run_vcr_example(options, examlpe)
    if options.fetch(:record).equal?(:skip)
      VCR.turned_off(&example)
    else
      name = extract_name(example.metadata.fetch(:full_description))

      VCR.use_cassette(name, options, &example)
    end
  end

  private

  def extract_name(full_description)
    full_description
        .split(/\s+/, 2)
        .join('/')
        .underscore
        .gsub(/[.#?]/, '/')
        .gsub(%r(\/+), '/')
        .strip
        .tr(' ', '_')
        .gsub(%r([/_ ]$), '')
  end
end
```